src/main/java/com/microfocus/application/automation/tools/results/parser/ReportParserManager.java

```diff
@@ -31,45 +31,66 @@
 import com.microfocus.application.automation.tools.results.parser.jenkinsjunit.JenkinsJUnitReportParserImpl;
 import com.microfocus.application.automation.tools.results.parser.mavensurefire.MavenSureFireReportParserImpl;
 import com.microfocus.application.automation.tools.results.parser.nunit.NUnitReportParserImpl;
+import com.microfocus.application.automation.tools.results.parser.nunit3.NUnit3ReportParserImpl;
 import com.microfocus.application.automation.tools.results.parser.testngxml.TestNGXmlReportParserImpl;
 import com.microfocus.application.automation.tools.results.service.almentities.AlmTestSet;
+import com.microfocus.application.automation.tools.sse.sdk.Logger;
+import hudson.FilePath;
 
 public class ReportParserManager {
 	
-	private static ReportParserManager instance = new ReportParserManager();
-	
-	List<ReportParser> parserList = new ArrayList<ReportParser> ();
-	
-	private ReportParserManager() {
-		parserList.add(new JenkinsJUnitReportParserImpl());
-		parserList.add(new MavenSureFireReportParserImpl());
-		parserList.add(new TestNGXmlReportParserImpl());
-		parserList.add(new NUnitReportParserImpl());
-		parserList.add(new AntJUnitReportParserImpl());
-	}
-	
-	public static ReportParserManager getInstance() {
+	private static ReportParserManager instance;
+
+	private List<ReportParser> parserList;
+	private FilePath workspace;
+	private Logger logger;
+
+	private ReportParserManager() {}
+
+	public static ReportParserManager getInstance(FilePath workspace, Logger logger) {
+		if (instance == null) {
```

Comment:
The double-checked locking doesn't work properly in Java, you could either drop lazy initialization and use eager initialization or ensure that all instances of the instance are volatile.
More information at: https://www.javaworld.com/article/2074979/java-concurrency/double-checked-locking--clever--but-broken.html